23.0.0.7-beta staging (#3196)
OpenLiberty/blogs: posts/2023-07-11-23.0.0.7-beta.adoc

To use Jakarta Data, you start by defining an entity class that corresponds to your data. With relational databases, the entity class corresponds to a database table and the entity properties (public methods and fields of the entity class) generally correspond to the columns of the table. An entity class can be:

- annotated with `jakarta.persistence.Entity` and related annotations from Jakarta Persistence
- a Java class without entity annotations, in which case the primary key is inferred from an entity property named `id` or ending with `Id`.

> You define one or more repository interfaces for an entity, annotate those interfaces as `@Repository` and inject them into components via CDI. The Jakarta Data provider supplies the implementation of the repository interface for you.

Reviewer: ```suggestion
You define one or more repository interfaces for an entity, annotate those interfaces as `@Repository`, and inject them into components with CDI. The Jakarta Data provider supplies the implementation of the repository interface for you.
```

> Here's a simple entity, repository, and usage of it.

Reviewer: is this to say that the simple entity _is_ the repository, or a simple entity _and_ a repository? Would the following als be correct?
```suggestion
The following example shows a simple entity with a single repository interface.
```

Reviewer 2: Yes, this change is fine.

    boolean discount(long productId, float discountRate);

    // experimental annotatively-defined query (not part of Jakarta Data 1.0)
    @Filter(by = "productId")
    @Update(attr = "price", op = Operation.Add)
    boolean increasePrice(long prodId, float amount);
}

public class MyServlet extends HttpServlet {
    @Inject
    Products products;

    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
            throws ServletException, IOException {
        // Request only the first 20 results on a page, ordered by price, then name, then id:
        Pageable pageRequest = Pageable.size(20).sortBy(Sort.desc("price"), Sort.asc("name"), Sort.asc("id"));
        Page<Product> page1 = products.findByNameIgnoreCaseContains(searchFor, pageRequest);
    }
}
----

For more information, refer to the the link:https://github.com/jakartaee/data/blob/main/spec/src/main/asciidoc/repository.asciidoc[draft of the Jakarta Data specification] and the link:https://search.maven.org/remotecontent?filepath=jakarta/data/jakarta-data-api/1.0.0-b2/jakarta-data-api-1.0.0-b2-javadoc.jar[corresponding Javadoc].

> Your feedback is welcome on all of the Jakarta Data features and will be helpful as the specification develops further. Let us know what you think and/or be involved directly in the specification on link:https://github.com/jakartaee/data[github].

Reviewer: "features" here might seem as though there is more than one OL beta feature for Jakarta Data
```suggestion
We welcome your feedback on Jakarta Data and it will be helpful as the specification develops further. Let us know what you think, or get involved directly in the specification on link:https://github.com/jakartaee/data[GitHub].
```
